src/org/authInfo.ts

```diff
@@ -419,7 +419,48 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
             logger.debug(`error updating auth file for ${orgAuthInfo.getUsername()}`, error);
           }
         } catch (error) {
-          logger.error(`Error connecting to devhub ${hubAuthInfo.username}`, error);
+          if (error instanceof Error && error.name === 'SingleRecordQuery_NoRecords' && fields.orgId) {
+            // Not a scratch org owned by this hub. Check if it's a sandbox.
+            try {
+              const prodOrg = await Org.create({ aliasOrUsername: hubAuthInfo.username });
+              const sbxProcess = await prodOrg.querySandboxProcessByOrgId(fields.orgId);
+              logger.debug(`${fields.orgId} is a sandbox of ${hubAuthInfo.username}`);
+
+              try {
+                await orgAuthInfo.save({
+                  ...fields,
+                  isScratch: false,
+                  isSandbox: true,
+                });
+              } catch (err) {
+                logger.debug(`error updating auth file for: ${orgAuthInfo.getUsername()}`, err);
+              }
+
+              try {
```

Comment:
would anything bad happen if you failed to update the AuthFile but then updated the SandboxConfig?

Reply:
Good question.  That's most likely a bad state though so probably just abort the mission (`return;`) if there's an error updating the auth file.  I'll make that change.